src/weppy.ts

```diff
@@ -0,0 +1,478 @@
+/// <reference path="../weppy.d.ts"/>
+
+// these seem not to be defined in window
+interface Window {
+	XMLHttpRequest?: XMLHttpRequest;
+	XDomainRequest?: XMLHttpRequest;
+	JSON?: JSON;
+}
+
+module WeppyImpl {
+
+	var protocolVersion = 3,
+		pathDelimiter = '.',
+		namespaceDelimiter = '::',
+		active = false,
+		options: WeppySettings = {
+			"host": '/weppy',
+			"transport": 'url', // 'url' or 'post'
+			"active": true,
+			"sample": 0.01,
+			"aggregationInterval": 1000,
+			"maxInterval": 5000,
+			"decimalPrecision": 3,
+			"page": 'index',
+			"context": {},
+			"debug": false
+		},
+		initTime = +(new Date),
+		queue, aggregationTimeout, maxTimeout, sentPerformanceData,
+		now = () => {
+			return window.performance && window.performance.now ? window.performance.now() : +(new Date);
+		},
+		log: any = () => {
+			if (options.debug) {
+				if (typeof options.debug == 'function') {
+					options.debug.apply(window, arguments);
+				} else {
+					window.console && window.console.log && window.console.log.apply
+						? window.console.log.apply(window.console, arguments) : void 0;
+				}
+			}
+		},
+		logError: any = () => {
+			window.console && window.console.error && window.console.error.apply &&
+				window.console.error.apply(window.console, arguments);
+		};
+
+	function round (num: number, precision = options.decimalPrecision): number {
+		return Math.round(num * Math.pow(10, precision)) / Math.pow(10, precision)
+	}
+
+	function buildPath (path: string, subpath: string, glue: string = pathDelimiter) {
+		return path + ( (path != '' && subpath != '') ? glue : '' ) + subpath;
+	}
+
+	function updateActive () {
+		active = options.active && Math.random() < options.sample;
+	}
+
+	function extend (first: any, second: any) {
+		var key;
+		if (first && second) {
+			for (key in second) {
+				if (second.hasOwnProperty(key)) {
+					first[key] = second[key];
+				}
+			}
+		}
+		return first || second;
+	}
+
+	function sortedJson (obj) {
+		var keys = Object.keys(obj).sort(),
+			i, ret = {};
+		for (i = 0; i < keys.length; i++) {
+			ret[keys[i]] = obj[keys[i]];
+		}
+		return JSON.stringify(ret);
+	}
+
+	export enum MetricType { Counter, Gauge, Timer }
+
+	module QueueData {
+		export interface Root {
+			[key: string]: Node;
+		}
+		export interface Node {
+			annotated?: AnnotatedKey;
+			data?: Data;
+		}
+		export interface AnnotatedKey {
+			[key: string]: AnnotatedValue;
+		}
+		export interface AnnotatedValue {
+			[key: string]: Node;
+			[key: number]: Node;
+		}
+		export interface Data {
+			value: number;
+			count?: number;
+			rollingAverage?: boolean;
+		}
+	}
+	class Queue {
```

Comment:
I think since the full code base is just 500 lines there's no reason to split it into multiple files.